x-pack/platform/plugins/shared/cases/server/cases_analytics/tasks/synchronization_task/synchronization_task_runner.test.ts

```diff
@@ -346,6 +325,53 @@ describe('SynchronizationTaskRunner', () => {
     });
   });
 
+  it('returns the previous state if the previous task is still running', async () => {
+    esClient.tasks.get.mockResolvedValueOnce({
+      completed: false,
+      task: {} as TasksTaskInfo,
+    });
+
+    const getESClient = async () => esClient;
+
+    taskRunner = new SynchronizationTaskRunner({
+      logger,
+      getESClient,
+      taskInstance,
+      analyticsConfig,
+    });
+
+    const result = await taskRunner.run();
+
+    expect(esClient.reindex).not.toBeCalled();
+    expect(result).toEqual({
+      state: taskInstance.state,
+    });
+  });
+
+  it('skips execution if the index does not exist', async () => {
+    esClient.indices.exists.mockResolvedValueOnce(false);
+
+    const getESClient = async () => esClient;
+
+    taskRunner = new SynchronizationTaskRunner({
+      logger,
+      getESClient,
+      taskInstance,
+      analyticsConfig,
+    });
+
+    const result = await taskRunner.run();
+
+    expect(esClient.reindex).not.toBeCalled();
+    expect(result).toBe(undefined);
```

Comment:
nit: Let's also check that `esClient.cluster.health` did not get called (it was the one that was creating the issues in our logs)